packages/@react-aria/overlays/src/Overlay.tsx

```diff
@@ -23,7 +23,11 @@ export interface OverlayProps {
    */
   portalContainer?: Element,
   /** The overlay to render in the portal. */
-  children: ReactNode
+  children: ReactNode,
+  /** Whether to contain focus within the overlay. This is an override, by default Dialogs contain and nothing else does. */
+  shouldContainFocus?: boolean,
+  /** Whether to restore focus when the FocusScope unmounts. */
+  shouldRestoreFocus?: boolean
```

Comment:
Maybe something like `disableFocusManagement`, and it could just not wrap the children in a focus scope at all? Then it could be used for other cases where users want to bring their own focus management solution.